web/react/utils/markdown.jsx

```diff
@@ -39,7 +40,14 @@ export class MattermostMarkdownRenderer extends marked.Renderer {
         if (title) {
             output += ' title="' + title + '"';
         }
-        output += ' target="_blank">' + text + '</a>';
+
+        if (outHref.lastIndexOf(Utils.getTeamURLFromAddressBar()) === 0) {
```

Comment:
`if (outHref.lastIndexOf(Utils.getTeamURLFromAddressBar()) === 0) {`
should be 
`if (outHref.lastIndexOf(Utils.getTeamURLFromAddressBar(), 0) === 0) {`
so that it just checks the start of the string and doesn't search through the whole thing like it would with indexOf